### PR TITLE
feat: Add "How we made Cron Monitoring reliable"

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: data/blog/how-we-made-cron-monitoring-reliable.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
+title: 'How we made Cron Monitoring reliable'
+date: '2023-11-14'
+tags: ['crons', 'database', 'infrastructure']
+draft: false
+summary: 'Having just surpassed over 7 million check-ins per day, let’s talk about what it’s taken to take to go from prototype to production ready cron monitoring.'
```
Comment (JoshFerge) [inline data/blog/how-we-made-cron-monitoring-reliable.md:6]: nit: passive voice
```suggestion
summary: 'We just surpassed over 7 million check-ins per day. Let's talk about what it took to go from prototype to production ready cron monitoring'
```

File: data/blog/scaling-cron-monitoring.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+layout: PostLayout
+canonicalUrl: scaling-cron-monitoring
+authors: ['evanpurkhiser']
+---
+
+Over the past year, we’ve been building out **Crons.** A feature that started as a [small prototype](https://github.com/getsentry/sentry/commit/d27ecf09c3) which we’ve turned into a a fully-fledged feature of Sentry’s Error, Performance, and Reliability monitoring tools. While the feature itself is still in a late-stage beta, we are already accepting **over 7 million check-ins** per day. With volume like that, we’ve had to make careful decisions about the scaling architecture, specifically around how we can guarantee reliability.
+
+# Early implementation
+
+If you’re unfamiliar with Sentry’s new Cron Monitoring feature, it is a way to validate and be alerted in scenarios where your scheduled jobs (e.g. classic cronjobs, celery beat tasks, Sidekiq, Lavaravel Scheduled Tasks, or even systemd timers) are operating as usual or fail to execute. This works by your scheduled task notifying us via “check-in” events.
```
Comment (mikejihbe) [inline data/blog/scaling-cron-monitoring.md:17]: Lavaravel -> Laravel

File: data/blog/scaling-cron-monitoring.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+- **Missed:** Failing to check-in at the expected schedule will create a “missed” check-in.
+- **Timed-Out:** Failing to complete an `in_progress` check-in will mark it as timed-out.
+- **Error:** Sending a `status=error` check-in will mark it as explicitly errored.
+
+![Sentry Cron Monitoring](/images/scaling-cron-monitoring/demo.png)
+
+In our initial prototype check-in ingestion was handled by a few API endpoints built into our Django application. Keeping things simple, check-ins are simply recorded as entries in a Postgres table. To detect missed check-ins we simply ran a celery beat task once per minute to look for monitors that did not have a check-in at the expected time. Detecting time-outs works similarly.
```
Comment (mikejihbe) [inline data/blog/scaling-cron-monitoring.md:25]: context on what the drawbacks were of this approach would help clarity/continuity

**Review by mikejihbe — APPROVED** [review]
Minor notes, but great post!

File: data/blog/scaling-cron-monitoring.md (hunk 3 above)
Comment (evanpurkhiser) [reply data/blog/scaling-cron-monitoring.md:25]: I think the next paragraph covers this well, what do you think @mikejihbe ?

**Pull request merged**